inotify.go

```diff
@@ -149,20 +183,31 @@ func (w *Watcher) readEvents() {
 			continue
 		}
 
+		// syscall.Read might have been woken up by Close. If so, we're done.
+		if w.isClosed() {
+			return
+		}
+
 		// If EOF is received
 		if n == 0 {
-			syscall.Close(w.fd)
-			close(w.Events)
-			close(w.Errors)
+			close(w.done)
 			return
 		}
 
 		if n < 0 {
-			w.Errors <- os.NewSyscallError("read", errno)
+			select {
```

Comment:
Yeah. This solves the problem of when someone Closes the watcher and moves on, but there is still an error waiting to be sent. Without those selects, it would block forever. Which is, in fact, what was happening in a lot of cases.